File: Beecrowd/1069-Diamantes-e-Areia/JAVA/Main.java
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		int n = 0,
		    esq = 0,
		    dir = 0,
		    diamantes = 0;
		String str;

		n = sc.nextInt(); // Ler quantidade de entradas
		sc.nextLine(); // Ler \n (quebra de linha)
			
		for (int i = 0; i < n; i++) {
			str = sc.nextLine(); // Ler entrada
			for (int j = 0; j < str.length(); j++) { // Passar por cada caractere da string
				// Verifica se é a abertura com o caractere menor
				if (str.charAt(j) == '<') {
					esq++;
				}
				// Verifica se é o fechamento com o caractere maior
				else if (str.charAt(j) == '>') {
					dir++;
				}
			}

			// Verificar quantidade de diamantes
			while (esq > 0 && dir > 0) {
				diamantes++;
				esq--; dir--;
			}

			System.out.println(diamantes);

			diamantes = 0; esq = 0; dir = 0; // Zerar variáveis
		}

		sc.close();
	}
}

File: Beecrowd/1069-Diamantes-e-Areia/CPP/Main.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
	int n = 0,
	    esq = 0,
	    dir = 0,
	    diamantes = 0;

	string str;
	cin >> n;

	for (int i = 0; i < n; i++) {
		cin >> str;
		for (char letra : str) { // Laço For Each para ler toda a string
			// Verifica abertura do caractere menor
			if (letra == '<') { 
				esq++;
			}	
			// Verifica fechamento do caractere maior
			else if (letra == '>') {
				dir++;
			}
		}
		
		// Contar quantidade de diamantes
		while (esq > 0 && dir > 0) {
			diamantes++;
			esq--; dir--;
		}

		cout << diamantes << endl;

		diamantes = 0; esq = 0; dir = 0; // Zera variáveis inteiras
	}

	return 0;
}
